### PR TITLE
runner: inline `call_runtest_hook`

### PR DESCRIPTION
**Pull request by bluetech**
- Reduce the common stacktrace by an entry - this is mostly for benefit of devs looking at crash logs.

- Reduce 6 slow `ihook` calls per test to 3.

Replaces #11851.

### CODE REVIEW

File: src/_pytest/runner.py [hunk 1]
```diff
@@ -223,13 +223,26 @@ def pytest_report_teststatus(report: BaseReport) -> Optional[Tuple[str, str, str
 def call_and_report(
     item: Item, when: Literal["setup", "call", "teardown"], log: bool = True, **kwds
```
Comment (nicoddemus) [inline src/_pytest/runner.py:224]: I believe we can further reduce `item.ihook` calls by also passing `ihook` as parameter? Then the caller (`runtestprotocol`) can get the `ihook` only once.

Comment (bluetech) [reply src/_pytest/runner.py:224]: I don't like passing `item.ihook` along with `item`, feels wrong...

**Review by nicoddemus — APPROVED** [review]

File: src/_pytest/runner.py (hunk 1 above)
Comment (nicoddemus) [reply src/_pytest/runner.py:224]: I can see that, OK then. 👍

Comment (RonnyPfannschmidt) [reply src/_pytest/runner.py:224]: I agree that it feels wrong, still if it brings measurable perf improvements it may be worth the structure

Comment (bluetech) [reply src/_pytest/runner.py:224]: I might do it next time I look at a profile :)

**Pull request merged**